test/tape/pool.js

```diff
@@ -101,7 +101,16 @@ test('#2321 dead connections are not evicted from pool', async (t) => {
           });
         })
       );
-      await delay(50); // wait driver to notice connection errors (2ms was enough locally)
+
+      // TODO: It looks like this test case can trigger the race condition
+      //       outlined in this issue comment:
+      //
+      //         https://github.com/knex/knex/issues/3636#issuecomment-592005391
+      //
+      //       For now, we're working around this problem by introducing a
+      //       1 second delay.  But, we should revisit this once the connection
+      //       pool logic has been refactored.
+      await delay(1000); // wait driver to notice connection errors (2ms was enough locally)
```

Comment:
Technically, this change really shouldn't be part of this PR.  But, the race condition caused the CI to fail, and this seemed like the easiest way to kick things off again.  (Related note: is there a re-run button in Travis-CI?  I haven't been able to find it)

Reply:
nvm -- just found the button in Travis-CI.  I just had to login before it appeared.